connector/google/google.go

```diff
@@ -252,34 +263,95 @@ func (c *googleConnector) createIdentity(ctx context.Context, identity connector
 	return identity, nil
 }
 
+func (c *googleConnector) getAllGroups(ctx context.Context, userKey string) ([]string, error) {
+	directMemberships, err := c.adminSrv.Groups.List().
+		UserKey(userKey).
+		Domain(c.domain).
+		Context(ctx).
+		Do()
+	if err != nil {
+		return nil, err
+	}
+
+	var groups []string
+	var groupsCh = make(chan string)
+	checkedGroups := sync.Map{}
+	g, cctx := errgroup.WithContext(ctx)
+
+	for _, group := range directMemberships.Groups {
```

Comment:
Is there a limit for concurrent requests to Google API? Can a concurrent request be rejected because of too many requests?

Reply:
Acoording to the [docs](https://developers.google.com/admin-sdk/directory/v1/limits) the quota is 3000 requests/100s and can be raised in the developer console. It doesn't look like the generated googleapis clients handle 403/userRateLimitExceeded, quotaExceeded or 429/rateLimitExceeded out of the box. We could potentially provide a hook for the user to provide a limit on in-flight requests through the `errgroup.Group`'s `SetLimit()` method.